Sources/NIOTransportServices/NIOTSSingletons.swift

```diff
@@ -0,0 +1,88 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if canImport(Network)
+import NIOCore
+
+@available(OSX 10.14, iOS 12.0, tvOS 12.0, watchOS 6.0, *)
+extension NIOTSEventLoopGroup {
+    /// A globally shared, lazily initialized, singleton ``NIOTSEventLoopGroup``.
+    ///
+    /// SwiftNIO allows and encourages the precise management of all operating system resources such as threads and file descriptors.
+    /// Certain resources (such as the main `EventLoopGroup`) however are usually globally shared across the program. This means
+    /// that many programs have to carry around an `EventLoopGroup` despite the fact they don't require the ability to fully return
+    /// all the operating resources which would imply shutting down the `EventLoopGroup`. This type is the global handle for singleton
+    /// resources that applications (and some libraries) can use to obtain never-shut-down singleton resources.
+    ///
+    /// Programs and libraries that do not use these singletons will not incur extra resource usage, these resources are lazily initialized on
+    /// first use.
+    ///
+    /// The loop count of this group is determined by `NIOSingletons.groupLoopCountSuggestion`.
+    ///
+    /// - note: Users who do not want any code to spawn global singleton resources may set
+    ///         `NIOSingletons.singletonsEnabledSuggestion` to `false` which will lead to a forced crash
+    ///         if any code attempts to use the global singletons.
+    public static var singleton: NIOTSEventLoopGroup {
+        return NIOSingletons.transportServicesEventLoopGroup
+    }
+}
+
+@available(OSX 10.14, iOS 12.0, tvOS 12.0, watchOS 6.0, *)
+extension EventLoopGroup where Self == NIOTSEventLoopGroup {
```

Comment:
Does this extension ever exist where the other does not?

Reply:
@glbrntt suggested this one. The answer is yes, in the many places where we take `any EventLoopGroup`. there you can now type `.singletonMultiThreadedEventLoopGroup`/`.singletonTransportServicesEventLoopGroup`.

I don't know if I love it but NIO released this pattern and even if we decided that it's bad, it's an easy one to deprecated. Given that it's in NIO we should also have it in TS.

Reply:
FWIW I think this is the more useful API for most users because it's easy to discover: IDEs will suggest this when you dot-tab-complete for `any EventLoopGroup`. If you didn't have this API users would have to know about the ELG types or singleton objects to find them.

Reply:
That's deeply weird.

Reply:
I think we have SwiftUI to thank for this 😃 e.g. https://developer.apple.com/documentation/swiftui/view/labelstyle(_:)